vnext/Microsoft.ReactNative/Utils/BatchingEventEmitter.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "JSValue.h"
+#include "ReactHost/React.h"
+#include "ReactPropertyBag.h"
+#include "winrt/Microsoft.ReactNative.h"
+
+#include <deque>
+#include <mutex>
+
+namespace winrt::Microsoft::ReactNative::implementation {
+struct BatchedEvent {
+  int64_t tag{};
+  winrt::hstring eventName;
+  JSValueObject eventObject;
```

Comment:
>JSValueObject [](start = 2, length = 13)

It should be JSValue. The JSValueObject is used as a 'builder' for JSValue objects. The JSValue is immutable and is good to pass values even between threads. The JSValueObject is mutable and typically used to create JSValue before we send it.